cmake/SetupCxxFlags.cmake

```diff
@@ -18,32 +18,48 @@ endif()
 # Always build with -g so we can view backtraces, etc. when production code
 # fails. This can be overridden by passing `-D DEBUG_SYMBOLS=OFF` to CMake
 if(${DEBUG_SYMBOLS})
-  set(CMAKE_CXX_FLAGS "-g ${CMAKE_CXX_FLAGS}")
+  set_property(TARGET SpectreFlags
+    APPEND PROPERTY INTERFACE_COMPILE_OPTIONS -g)
 endif(${DEBUG_SYMBOLS})
 
 # Always compile only for the current architecture. This can be overridden
 # by passing `-D OVERRIDE_ARCH=THE_ARCHITECTURE` to CMake
 if(NOT "${OVERRIDE_ARCH}" STREQUAL "OFF")
-  set(CMAKE_CXX_FLAGS "-march=${OVERRIDE_ARCH} ${CMAKE_CXX_FLAGS}")
+  set_property(TARGET SpectreFlags
```

Comment:
does it matter if these flags are being appended instead of prepended now?

Reply:
No, not anymore. Before we had `OVERRIDE_ARCH` it did, but that didn't work correctly anyway